src/main/java/uk/gov/hmcts/reform/civil/helpers/BundleRequestMapper.java

```diff
@@ -0,0 +1,279 @@
+                .courtLocation(caseData.getHearingLocation().getValue().getLabel())
+                .hearingDate(null != caseData.getHearingDate()
+                                 ? DateFormatHelper.formatLocalDate(caseData.getHearingDate(), "dd-MM-yyyy") : null)
+                .ccdCaseReference(caseData.getCcdCaseReference())
+                .build();
+        bundlingCaseData = mapRespondent2Applicant2Details(bundlingCaseData, caseData);
+        return  bundlingCaseData;
+    }
+
+    private BundlingCaseData mapRespondent2Applicant2Details(BundlingCaseData bundlingCaseData, CaseData caseData) {
+        if (null != caseData.getAddApplicant2() && caseData.getAddApplicant2().equals(YesOrNo.YES)) {
+            bundlingCaseData.toBuilder().hasApplicant2(true);
+        }
+        if (null != caseData.getAddRespondent2() && caseData.getAddRespondent2().equals(YesOrNo.YES)) {
+            bundlingCaseData.toBuilder().hasRespondant2(true);
+        }
+        if (null != caseData.getApplicant2()) {
+            bundlingCaseData.toBuilder().applicant2(caseData.getApplicant2());
+        }
+        if (null != caseData.getRespondent2()) {
+            bundlingCaseData.toBuilder().respondent2(caseData.getRespondent2());
+        }
+        return bundlingCaseData;
+    }
+
+    private ServedDocument mapServedDocuments(ServedDocumentFiles servedDocumentFiles) {
+        List<BundlingRequestDocument> bundlingServedDocFiles = new ArrayList<>();
+        if (!Optional.ofNullable(servedDocumentFiles).isEmpty() && null != servedDocumentFiles.getParticularsOfClaimDocument()) {
+            servedDocumentFiles.getParticularsOfClaimDocument().forEach(document -> {
+                bundlingServedDocFiles.add(BundlingRequestDocument.builder()
+                                               .documentFileName(document.getValue().getDocumentFileName())
+                                               .documentLink(DocumentLink.builder()
+                                                                 .documentUrl(document.getValue().getDocumentUrl())
+                                                                 .documentBinaryUrl(document.getValue().getDocumentBinaryUrl())
+                                                                 .documentFilename(document.getValue().getDocumentFileName()).build())
+                                               .build());
+
+            });
+        }
+        List<Element<BundlingRequestDocument>> particulars = ElementUtils.wrapElements(bundlingServedDocFiles);
+        return  ServedDocument.builder().particularsOfClaimDocument(particulars).build();
+    }
+
+    private List<Element<BundlingRequestDocument>> mapSystemGeneratedcaseDocument(List<Element<CaseDocument>> systemGeneratedCaseDocuments, Document orderSDODocumentDJ) {
+        List<BundlingRequestDocument> bundlingSystemGeneratedCaseDocs = new ArrayList<>();
+        if (!Optional.ofNullable(systemGeneratedCaseDocuments).isEmpty()) {
```

Comment:
Short-circuit this "if" statement.

Before you do however, add a test around the missing branch (that is: where `systemGeneratedCaseDocuments` is empty).

Reply:
Marking as resolved as at this stage I don't necessarily want to change the code.